lib/rubygems/ext/builder.rb

```diff
@@ -221,9 +203,10 @@ def build_extensions
     FileUtils.rm_f @spec.gem_build_complete_path
 
     @spec.extensions.each do |extension|
-      break if @ran_rake
+      builder = builder_for extension
+      build_extension extension, builder, dest_path
 
-      build_extension extension, dest_path
+      break if builder.is_a? Gem::Ext::RakeBuilder
```

Comment:
Code breaks after rake builder is triggered for some reason. I have no idea why this is needed. All spec passes also without this. It was added at https://github.com/rubygems/rubygems/commit/20cdf22b844a63d163cc80eff819b99330bd6fa1.

In theory if you have multiple extensions defined in gemspec, if rake one is specified, no other extension one after rake will be build.

:thinking: I don't see any reason for this. But for now I have kept this to not modify any behaviour.

Reply:
Haven't been following that closely, but...

> In theory if you have multiple extensions defined in gemspec

An example would be EventMachine, two extconf.rb files, and two *.so files compiled.

Reply:
As I see https://github.com/eventmachine/eventmachine/blob/d761d2a6916b854b5d62aeaa4775be359e9cf6ce/eventmachine.gemspec#L16 there are two extensions defined, but both using `extconf` builder. But if first one will use rake builder, the second one will not be built and I have no idea what's the reason for this behaviour.